modules/eks/karpenter/README.md

```diff
@@ -281,6 +284,31 @@ For your cluster, you will need to review the following configurations for the K
       ttl_seconds_until_expired: 2592000
     ```
 
+## Node Interruption
+
+Karpenter also supports Node Interruption. If interruption-handling is enabled, Karpenter will watch for upcoming involuntary interruption events that would cause disruption to your workloads. These interruption events include:
+
+- Spot Interruption Warnings
+- Scheduled Change Health Events (Maintenance Events)
+- Instance Terminating Events
+- Instance Stopping Events
+
+:::info
+
+The Node Interruption Handler is not the same as the Node Termination Handler. The latter works fine and cleanly shuts down the node in 2 minutes. The former gets advance notice, so it can have 5-10 minutes to shut down a node.
+
+:::
+
+For more details, see refer to [Karpenter docs](https://karpenter.sh/v0.27.5/concepts/deprovisioning/#interruption)
+
+To enable Node Interruption, set `var.interruption_handler_enabled` to `true`. This will enable a SQS queue and a set of Event Bridge rules to handle interruption with Karpenter.
+
+## Custom Resource Definition (CRD) Management
+
+Karpenter ships with a few Custom Resource Definitions (CRDs). If you bump the version of the `karpenter` helm chart, CRDs may not be upgraded at the same time, requiring manual steps to upgrade CRDs after deploying the latest chart. However Karpenter now supports an additional, independent helm chart for CRD management. This helm chart, `karpenter-crd`, can be installed alongside the `karpenter` helm chart to automatically manage the lifecycle of these CRDs.
+
+To deploy the `karpenter-crd` helm chart, set `var.crd_chart_enabled` to `true`.
+
```

Comment:
If you have an existing deployment where the CRDs were not deployed by the `karpenter-crd` helm chart, then you want to use the supplied `karpenter-crd-upgrade` script first, to update the CRDs so that Helm will take over management of them. See [Helm error when installing the Karpenter CRD chart](https://karpenter.sh/v0.31/troubleshooting/#helm-error-when-installing-the-karpenter-crd-chart) for details. The `karpenter-crd-upgrade` script applies both fixes described there.